probert/raid.py

```diff
@@ -117,15 +116,32 @@ def probe(context=None, report=False):
 
     raids = {}
     for device in sane_block_devices(context):
-        if 'MD_NAME' in device and device.get('DEVTYPE') == 'disk':
-            devname = device['DEVNAME']
-            attrs = udev_get_attributes(device)
-            attrs['size'] = str(read_sys_block_size_bytes(devname))
+        if device.get('DEVTYPE') != 'disk':
+            continue
+        devname = device['DEVNAME']
+        if 'MD_NAME' in device or device.get('MD_METADATA') == 'imsm':
             devices, spares = get_mdadm_array_members(devname, device)
             cfg = dict(device)
-            cfg.update({'raidlevel': device['MD_LEVEL'],
-                        'devices': devices,
-                        'spare_devices': spares})
+            if device.get('MD_METADATA') == 'imsm':
+                # All disks in a imsm container show up as spares in --detail
```

Comment:
I did find this upon some web searching - is it possible the imsm scenario is unhappy about something?
https://superuser.com/questions/1328814/disks-have-become-spare-with-fake-raid-is-there-any-change-to-remount-the-volum
Note that in the commit example, the size in the listed output is '0', but only for the imsm case.

Put differently, would this be a problem if a user environment imsm case reported the devices as devices and not spares?